### PR TITLE
add support for mongo connection strings that are to a cluster, added…

### LINKED ISSUE #860: ERR_INVALID_URL: Invalid url after newrelic 8.2.0 update
#  Invalid URL error for mongodb requests

## Description

# 
Getting following error while making any API call consisting of mongodb call
```
{
  "message": "Invalid URL: mongodb://<user>:<pass>@10.0.1.20:27017,10.0.1.21:27017,10.0.1.22:27017/<dbname>?replicaSet=re_common_set&readPreference=secondaryPreferred",
  "code": "ERR_INVALID_URL",
  "sentry": "<sentry>"
}
```

Stacktrace: 
```

internal/url.js in onParseError at line 259:9
internal/url.js in new URL at line 335:5
/project/node_modules/newrelic/lib/instrumentation/mongodb/common.js in getInstanceAttributeParameters at line 141:36

/project/node_modules/newrelic/lib/instrumentation/mongodb/common.js in Collection.queryDescFunc at line 80:24

/project/node_modules/newrelic/lib/shim/datastore-shim.js in Collection.queryRecord at line 725:29

/project/node_modules/newrelic/lib/shim/shim.js in Collection.wrapper at line 929:33

/project/node_modules/@sentry/tracing/dist/integrations/mongo.js in Collection.<anonymous> at line 138:34

/project/node_modules/mongoose/lib/drivers/node-mongodb-native/collection.js in NativeCollection.<computed> [as findOne] at line 191:33

/project/node_modules/mquery/lib/collection/node.js in NodeCollection.findOne at line 40:19

/project/node_modules/mquery/lib/mquery.js in model.Query.Query.findOne at line 2024:20

/project/node_modules/mongoose/lib/query.js in model.Query.<anonymous> at line 2360:22

/project/node_modules/mongoose/lib/helpers/query/wrapThunk.js in model.Query._wrappedThunk [as _findOne] at line 27:8

/project/node_modules/kareem/index.js in null.<anonymous> at line 370:33

/project/node_modules/newrelic/lib/shim/shim.js in Shim.applySegment at line 1430:20

/project/node_modules/newrelic/lib/shim/shim.js in wrapper at line 2083:17

internal/process/task_queues.js in processTicksAndRejections at line 75:11
```

## Expected Behavior

there should not be any unnecessary error while making requests

## Steps to Reproduce

Use newrelic latest version to tracing

## Your Environment

Node Version: v14.16.1
newrelic: ^8.2.0
Sentry: ^6.11.0

## Additional context

No

### PR DESCRIPTION
**Pull request by bizob2828**
## Proposed Release Notes
 * Added support for properly setting the `host` and `port` for mongodb requests that are to cluster.
 
## Links
Closes #860 

## Details
I added 2 additional suites of tests for a cluster configuration. One with host/port and one with domain sockets.  

**Note**: To run the domain socket tests you have to have `mongod` on your local machine and run locally before running

```
NPM7=1 npm run versioned-tests mongodb
```
Also, if you use a newer version of `mongod` it will fail on some tests because it assumes you're running v2 so some commands may not be available.

### CODE REVIEW

File: lib/instrumentation/mongodb/common.js [hunk 1]
```diff
@@ -97,25 +96,57 @@ common.captureAttributesOnStarted = function captureAttributesOnStarted(shim, in
...
+/**
+ * Extracts the host and port from a connection string
+ * This also handles if connection string is a domain socket
+ * Mongo sticks the path to the domain socket in the "host" slot, but we
+ * want it in the "port", so if we have a domain socket we need to change
+ * the order of our parameters.
+ *
+ * @param {Shim} shim
+ * @param {string} connStr
+ * @param {string} db database name
```
Comment (carlo-808) [inline lib/instrumentation/mongodb/common.js:125]: missed documenting the `client` param here.

File: package-lock.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
   "packages": {
     "": {
       "name": "newrelic",
-      "version": "8.1.0",
+      "version": "8.2.0",
```
Comment (carlo-808) [inline package-lock.json:9]: odd that wasn't updated with the release.

**Review by carlo-808 — COMMENTED** [review]
Looks good overall. Nice break out of connection instrumentation functions based on mongodb version. I just had one minor comment. I'm curious about the `mongod` you mentioned in PR notes. It looks like we skip the domain socket tests if mongod is not installed? Should it be installed as part of versioned tests?

File: package-lock.json (hunk 2 above)
Comment (bizob2828) [reply package-lock.json:9]: it did, not sure if this was because I didn't rebase properly? https://github.com/newrelic/node-newrelic/blob/main/package-lock.json#L3

**Comment by bizob2828** [comment]
> Looks good overall. Nice break out of connection instrumentation functions based on mongodb version. I just had one minor comment. I'm curious about the `mongod` you mentioned in PR notes. It looks like we skip the domain socket tests if mongod is not installed? Should it be installed as part of versioned tests?

Good question. I can log a ticket and we can groom.  Personally I think we should do this in CI because it does creates unique edge cases that need fixed at times.

**Review by carlo-808 — APPROVED** [review]
👍
